scripts/build-xml

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+
+set -e
+
+# cli args
+in_dir='src/tiler/src/config/'
+out_dir='transpiled-configs/'
+s3_bucket=''
+node_env='production'
+
+function usage() {
+    echo -n "Usage: $(basename "${0}") -i <path> -o <path>
+Transpiles a directory's worth of MML+MSS files into Tilegarden-readable XML
+Options:
+    -h, --help          Display this help text
+    -i, --in <path>     Relative path to input directory [${in_dir}]
+    -o, --out <path>    Relative path to output directory [${out_dir}]
+    -b, --bucket <name> Uploads XML to the specified S3 bucket, instead of saving to disk (requires AWS CLI installed on your machine, uses credentials from .env)
+    -d, --development   Build using development variables, instead of production ones
+"
+    exit 0
+}
+
+function main() {
+    # if s3_bucket has been set, send all files to a temporary directory
+    # rather than the default out_dir
+    if [ ! -z ${s3_bucket} ]; then
+        out_dir=$(mktemp -d)
+    fi
+
+    # transpile files and save in the out_dir directory
+    for file in $(echo "${in_dir}/*")
+    do
+        ext="${file##*.}"
+        if [ "$ext" == "mml" ]; then
+            # get output path
+            filename="${file##*/}"
+            base="${filename%%.*}"
+            outPath="${out_dir}/${base}.xml"
+
+            mkdir -p "${out_dir}"
+
+            echo "Transpiling ${file} => ${outPath}"
+            docker-compose run -e "NODE_ENV=${node_env}" -v "$PWD/${in_dir}:/home/tiler/configs" tiler ./scripts/build-xml.sh "/home/tiler/configs/${base}.mml" "$(< ${file})" > ${outPath}
+        fi
+    done
+
+    # Upload files to s3 if told to
+    if [ ! -z ${s3_bucket} ]; then
+        if [ ! -x "$(command -v aws)" ]; then
+            echo "ERROR: cannot upload to S3, AWS CLI not installed!"
+            exit 1
+        fi
+
+        s3_url="s3://${s3_bucket}/"
+        echo "Uploading files to ${s3_url}..."
+
+        # Load AWS profile settings from env, if possible
+        if [[ -f .env ]]; then
+            set -a
+            source ./.env
+            set +a
+        fi
+
+        for config in $(echo "${out_dir}/*"); do
+            aws s3 cp "${config}" ${s3_url} --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers
```

Comment:
Was the Lambda function not able to read the config files without this?  This appears to be granting global (including public) read access, which is not what we want.  I was able to load https://s3.amazonaws.com/tilegarden-dev/map-config.xml in an anonymous browser window.

Reply:
To be honest I forgot why I did this, but thanks for pointing it out because for sure these files shouldn't be publicly accessible.